hazelcast/src/main/java/com/hazelcast/cluster/impl/ClusterServiceImpl.java

```diff
@@ -667,6 +668,11 @@ public void handleJoinRequest(JoinRequest joinRequest, Connection connection) {
             if (now > timeToStartJoin) {
                 startJoin();
             }
+            if (node.isMaster()) {
+                long startTime = clusterClock.getClusterStartTime();
+                nodeEngine.getOperationService().send(new ClusterStartTimeUpdateOperation(startTime),
```

Comment:
I have updated send with invokeOnTarget, so this operation will be tried DEFAULT_TRY_COUNT times. Even if it fails after all tries, that particular node will not set its clusterStartTime and clusterId correctly. This process will be tried again when a new node joins cluster. Also I created a new operation instead of using FinalizeJoinOperation because FinalizeJoinOperation is used more critical sections of cluster forming. I did not understand the mechanisms where it is used. So, I created another independent operation.